pullRequests.go fails with panic: runtime error: index out of range [0] with length 0

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Creation of Pull Request fails in `Apply` phase with `runtime error: index out of range [0] with length 0`. See log output:

```
PULL REQUESTS
=============

panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/updatecli/updatecli/pkg/core/pipeline.(*Pipeline).RunPullRequests(0xc000c40370)
	/home/runner/work/updatecli/updatecli/pkg/core/pipeline/pullRequests.go:48 +0x1b37
github.com/updatecli/updatecli/pkg/core/pipeline.(*Pipeline).Run(0xc000c40370)
	/home/runner/work/updatecli/updatecli/pkg/core/pipeline/main.go:234 +0x3ce
github.com/updatecli/updatecli/pkg/core/engine.(*Engine).Run(0x3c565e0)
	/home/runner/work/updatecli/updatecli/pkg/core/engine/main.go:250 +0x57c
github.com/updatecli/updatecli/cmd.run({0x246dd90?, 0x0?})
	/home/runner/work/updatecli/updatecli/cmd/root.go:89 +0x3c6
github.com/updatecli/updatecli/cmd.glob..func1(0x3c2cf20?, {0x246c652?, 0x4?, 0x4?})
	/home/runner/work/updatecli/updatecli/cmd/apply.go:30 +0x11b
github.com/spf13/cobra.(*Command).execute(0x3c2cf20, {0xc000626440, 0x4, 0x4})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:876 +0x67b
github.com/spf13/cobra.(*Command).ExecuteC(0x3c2c7a0)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
github.com/updatecli/updatecli/cmd.Execute()
	/home/runner/work/updatecli/updatecli/cmd/root.go:42 +0x65
main.main()
	/home/runner/work/updatecli/updatecli/main.go:8 +0x17
```

### Expected Behavior

Action used to work fine with release `0.30.0` and started to fail with  `0.31.0` as well as  `0.32.0`.

### Steps To Reproduce

See https://github.com/jenkinsci/custom-folder-icon-plugin/actions/runs/2960947456

### Environment

```markdown
- OS: Ubuntu latest
- updatecli: v0.32.0 and v0.31.0
```


### Anything else?

Maybe I am missing something but the changelogs do not indicate any change that would cause my configuration to become malformed or incompatible.

--- a/e2e/updatecli.d/success.d/githubPullrequest.yaml
+++ b/e2e/updatecli.d/success.d/githubPullrequest.yaml
@@ -11,18 +11,15 @@ scms:
       branch: main
 
 sources:
-  jenkins: 
+  jenkins:
     name: Get updatecli config title
-    kind: yaml 
+    kind: jenkins
     scmid: default
-    spec:
-      file: "updatecli/updatecli.d/updatecli.yaml"
-      key: "title"
 
 targets:
-  jenkins: 
+  jenkins:
     name: Update updatecli config title
-    kind: yaml 
+    kind: yaml
     scmid: default
     spec:
       file: "updatecli/updatecli.d/updatecli.yaml"

--- a/e2e/updatecli.d/success.d/githubPullrequest.yaml
+++ b/e2e/updatecli.d/success.d/githubPullrequest.yaml
@@ -1,0 +1,34 @@
+name: Test Github Pullrequest
+
+scms:
+  default:
+    kind: github
+    spec:
+      owner: updatecli
+      repository: updatecli
+      username: '{{ requiredEnv "GITHUB_ACTOR"}}'
+      token: '{{ requiredEnv "GITHUB_TOKEN" }}'
+      branch: main
+
+sources:
+  jenkins: 
+    name: Get updatecli config title
+    kind: yaml 
+    scmid: default
+    spec:
+      file: "updatecli/updatecli.d/updatecli.yaml"
+      key: "title"
+
+targets:
+  jenkins: 
+    name: Update updatecli config title
+    kind: yaml 
+    scmid: default
+    spec:
+      file: "updatecli/updatecli.d/updatecli.yaml"
+      key: "title"
+
+pullrequests:
+  prID1:
+    scmid: default
+    kind: github

--- a/pkg/core/pipeline/pullRequests.go
+++ b/pkg/core/pipeline/pullRequests.go
@@ -21,7 +21,12 @@ func (p *Pipeline) RunPullRequests() error {
 	}
 
 	for id, pr := range p.PullRequests {
-		relatedTargets := p.SearchAssociatedTargetsID(pr.Config.ScmID)
+		relatedTargets, err := p.SearchAssociatedTargetsID(id)
+
+		if err != nil {
+			logrus.Errorf(err.Error())
+			continue
+		}
 
 		if _, ok := p.SCMs[pr.Config.ScmID]; !ok {
 			return fmt.Errorf("scm id %q couldn't be found", pr.Config.ScmID)
@@ -33,7 +38,7 @@ func (p *Pipeline) RunPullRequests() error {
 		pr = p.PullRequests[id]
 		pr.Config = p.Config.Spec.PullRequests[id]
 
-		err := pr.Update()
+		err = pr.Update()
 		if err != nil {
 			return err
 		}
@@ -172,9 +177,13 @@ func (p *Pipeline) GetTargetsIDByResult(targetIDs []string) (
 }
 
 // SearchAssociatedTargetsID search for targets related to a pullrequest based on a scm configuration
-func (p *Pipeline) SearchAssociatedTargetsID(prID string) []string {
+func (p *Pipeline) SearchAssociatedTargetsID(pullrequestID string) ([]string, error) {
 
-	scmid := p.PullRequests[prID].Config.ScmID
+	scmid := p.PullRequests[pullrequestID].Config.ScmID
+
+	if len(scmid) == 0 {
+		return []string{}, fmt.Errorf("scmid %q not found for pullrequest id %q", scmid, pullrequestID)
+	}
 	results := []string{}
 
 	for id, target := range p.Targets {
@@ -183,5 +192,5 @@ func (p *Pipeline) SearchAssociatedTargetsID(prID string) []string {
 		}
 	}
 
-	return results
+	return results, nil
 }

